src/box/vy_point_lookup.c

```diff
@@ -185,15 +185,11 @@ vy_point_lookup_scan_slices(struct vy_lsm *lsm, const struct vy_read_view **rv,
 							   ITER_EQ, key);
 	assert(range != NULL);
 	int slice_count = range->slice_count;
-	size_t size;
 	size_t region_svp = region_used(&fiber()->gc);
-	struct vy_slice **slices =
-		region_alloc_array(&fiber()->gc, typeof(slices[0]), slice_count,
-				   &size);
-	if (slices == NULL) {
-		diag_set(OutOfMemory, size, "region_alloc_array", "slices");
-		return -1;
-	}
+	struct vy_slice **slices = NULL;
+	if (slice_count > 0)
+		slices = xregion_alloc_array(&fiber()->gc, typeof(slices[0]),
+					     slice_count);
```

Comment:
If `slice_count` is 0 better return from this function immediately.

Reply:
Done.
```diff
@@ -186,10 +186,11 @@ vy_point_lookup_scan_slices(struct vy_lsm *lsm, const struct vy>
        assert(range != NULL);
        int slice_count = range->slice_count;
        size_t region_svp = region_used(&fiber()->gc);
-       struct vy_slice **slices = NULL;
-       if (slice_count > 0)
-               slices = xregion_alloc_array(&fiber()->gc, typeof(slices[0]),
-                                            slice_count);
+       if (slice_count == 0)
+               return 0;
+       struct vy_slice **slices =
+               xregion_alloc_array(&fiber()->gc, typeof(slices[0]),
+                                   slice_count);
        int i = 0;
        struct vy_slice *slice;
        rlist_foreach_entry(slice, &range->slices, in_range) {
```